Blog: May heartbeat (#3554)
iterative/dvc.org: content/blog/2022-05-16-may-22-heartbeat.md

---
title: May '22 Heartbeat
date: 2022-05-16
description: >
  Monthly updates are here! You will find a link to Chip Huyen's new book, great

>   guides and frameworks on the iterative nature of AI, tons of Company news,

Reviewer: ```suggestion
  guides and frameworks on the iterative nature of AI, tons of company news,
```

>   Dmitry on TFIR, beyond ml use cases and more! Welcome to May!

Reviewer: ```suggestion
  Dmitry on TFIR, beyond machine learning use cases and more! Welcome to May!
```

Reviewer: Spelling out machine learning helps with SEO.

picture: 2022-05-16/may-heartbeat.png
author: jeny_defigueiredo
commentsUrl: https://discuss.dvc.org/t/may-22-heartbeat/1191
tags:
  - Heartbeat
  - DVC
  - CML
  - TPI
  - Git
  - Modular Infrastructure
  - Mission Statement
  - Terraform Provider Iterative
  - Chip Huyen
  - TFIR
---

# AI/ML News

![Designing Machine Learning Systems](/uploads/images/2022-05-16/chip-huyen.jpeg 'Designing Machine Learning Systems :wrap-left ==200')

## Chip Huyen: Designing Machine Learning Systems

[**Chip Huyen**](https://www.linkedin.com/in/chiphuyen/) just came out with a
new book with [O'Reilly](https://oreilly.com) entitled
[Designing Machine Learning Systems](https://www.oreilly.com/library/view/designing-machine-learning/9781098107956/).  
I'm not going to pontificate here; Chip Huyen wrote it, the reviews are shining,
need I say more?

## Jenny Abramov: An Agile Framework for AI Projects — Development, QA, Deployment and Maintenance

[**Jenny Abramov**](https://www.linkedin.com/in/jennyabramov/)
[wrote a piece](https://towardsdatascience.com/an-agile-framework-for-ai-projects-development-cbe115ba86a2)
in [Toward Data Science](https://towardsdatascience.com/) with the purpose to

> present an "iterative-lifecycle framework," that is adapted to AI-centered

Reviewer: ```suggestion
present an iterative-lifecycle framework," that is adapted to AI-centered
```

software. She outlines important considerations as you work through the
framework that depends on your use case, data, and business problem.

She suggests using DVC for your larger, more complex datasets and also about the
need for reproducibility in experimentation with which DVC can help you
[(see Technical Product Manager, Dave Berenbaum’s post on experiment versioning.)](https://dvc.org/blog/ml-experiment-versioning)

> In addition, she discusses issues with quality assurance in Deployment and the

Reviewer: ```suggestion
In addition, she discusses issues with quality assurance in deployment and the
```

a frequent contributor to the MLOps Community in general, often driving
discussions and helping others in the
[MLOps Community Slack channel,](https://mlops-community.slack.com/join/shared_invite/zt-178s99cyv-Q~whRpqbhgMTBrOcbjnDIQ#/shared-invite/email)
posed an interesting point about code quality on LinkedIn. Join the discussion
and weigh in at this post:

<iframe src="https://www.linkedin.com/embed/feed/update/urn:li:share:6931541880090324992" height="800" width="504" frameborder="0" allowfullscreen="" title="Embedded post"></iframe>

# Company News

## ICYMI: We released TPI! 🎉

On April 27th we released the latest offering to our tool ecosystem.

https://media.giphy.com/media/ut7lqhIfOscbjuU6YQ/giphy.gif

[Terraform Provider Iterative (TPI)](https://tpi.cml.dev) is a Terraform plugin
built with machine learning in mind. Full lifecycle management of computing
resources (including GPUs and respawning spot instances) from several cloud
vendors (AWS, Azure, GCP, K8s)... without needing to be a cloud expert.

- **Lower cost with spot recovery:** transparent data checkpoint/restore &

>   auto-respawning of low-cost spot/preemptible instances

Reviewer: ```suggestion
  auto-respawning of low-cost spot/pre-emptible instances
```

Author: There's apparently not a hyphen in that word.  https://www.google.com/search?q=preemtible&oq=preemtible&aqs=chrome.0.69i59j0i10l9.5603j0j7&sourceid=chrome&ie=UTF-8

Weird to me too!

Reviewer: Huh! English is weird. 😅

- 🎥 [Watch the video](https://www.youtube.com/watch?v=2fEgO8SazSE&t=2s)
- 🪐
  [TPI with Jupyter Notebooks Repo](https://github.com/iterative/blog-tpi-jupyter)

Stay tuned for more tutorials and use cases to come!

https://media.giphy.com/media/MrCYIN3x0SgdG/giphy.gif

## 🚀~~Mission Impossible~~ - We have a mission statement!

We did it! This year we surveyed the entire team to arrive at a mission
statement for Iterative. It was no small feat to decide on what it should be
given the early stage of our industry, the variety of our tools, and always a
struggle - figuring out the best and most concise way to convey these ideas (you
know our penchant for abbreviations). But we persevered and succeeded. Behold
Iterative's new mission statement:

> We deliver the best developer experience for machine learning teams by
> creating an ecosystem of open, modular ML tools.

As always the door is open for your feedback on how we can serve your needs
better!

## ODSC East

We attended our first post-pandemic, in-person conference in Boston last month.
It was awesome to be together as a team, see
[**Dmitry Petrov**](https://twitter.com/FullStackML),

> [**Milicia Mc Gregor**](https://twitter.com/FlippedCoding), and

Reviewer: ```suggestion
[**Milecia McGregor**](https://twitter.com/FlippedCoding), and
```

Author: Thanks. I keep doing this! 🙈 I did this more than once I think.  I go through the find.

you join us for a Release Party Meetup. We will be celebrating the release of
the new addition to our open-source tool ecosystem and have a demo of said tool!
To join the fun,
[RSVP to the Meetup ](https://www.meetup.com/Machine-Learning-Engineer-Community-Virtual-Meetups/events/285789441/)
and mark your calendar!

<external-link
href="https://www.meetup.com/Machine-Learning-Engineer-Community-Virtual-Meetups/events/285789441/"
title="New Tool Release Party"
description="Join us May 25th. RSVP for New Tool Release Party!"
link="https://meetup.com"
image="/uploads/images/2022-05-16/release-party-meetup.png"/>

## New hires

[**Wolmir Nemitz**]() is our first team member from South America! We're getting
closer to covering all the continents on
[our remote team map](https://iterative.ai/about)! From Brazil, Wolmir joins us
as an Engineer for the 🤫 team (you'll find out June 14th). Wolmir has four
dogs, two tortoises, and a budgie! 🦜

[**Pavel Chekmaryov**](https://www.linkedin.com/in/ufijuice/) joins us in People
Operations, managing the hiring pipeline from Frankfurt, Germany, but soon to be
Canada! He has spent the last eight years in startups, most recently at OccurAI,

> reinventing recruitment in the deeptech/ML field. We look forward to him helping

Reviewer: ```suggestion
reinventing recruitment in the deep-tech/ML field. We look forward to him helping
```

to grow our amazing team!

## Open Positions

Even with our amazing new additions to the team, we're still hiring!
[Use this link](https://iterative.notion.site/Iterative-ai-is-Hiring-852cb978129645e1906e2c9a878a4d22)
to find details of all the positions and share with anyone you think may be
interested! 🚀

> ![Iterative.ai is Hiring](/uploads/images/2022-01-18/hiring.jpeg '=800')

Reviewer: This image isn't rendering.

Author: Interesting. I've used the same code for the past three Heartbeat.  I just moved a copy of the image over. 👍

_Iterative is Hiring
([Source link](https://iterative.notion.site/Iterative-ai-is-Hiring-852cb978129645e1906e2c9a878a4d22))_

# Community News

> ## Yet another tool comparison, Imagine that!

Reviewer: ```suggestion
## Yet another tool comparison, imagine that!
```

https://media.giphy.com/media/lWa7aAo62YZLwtk3nj/giphy.gif

So each month I tell you about yet another post to help you attempt to make
sense of the vast MLOps tool space. Well, this month is no different. I mean you
could be new here, right? 🤷🏽‍♀️ [DoltHub](https://dolthub.com) tries to bring some
clarity
[with this piece](https://www.dolthub.com/blog/2022-04-27-data-version-control/)
by comparing different data versioning tools and the intricacies of each. You do
your research. You know we're partial.

![Data Version Control tools](/uploads/images/2022-05-16/data-version-control.png '=800')
_Data Version control tools
([Source link](https://ml-ops.org/content/mlops-principles))_

I’m starting to wonder if all Data Science/AI teams need a role with the sole
responsibility of the job to keep up to date with all the new tooling and
changes/updates to existing tooling in the MLOps space and what might best work
for the team. What should this position be called? The best answer wins a DVC

> t-shirt. See this Twitter thread to answer. (Hint: Funny answers will likely win

Reviewer: Link to the Twitter thread?

Author: I just added it.

😉). Deadline: May 31st. Pass it around...

## Andrey Cheptsov: Notebooks and MLOps. Choose One.

[**Andrey Cheptsov**](https://www.linkedin.com/in/andrey-cheptsov/) writes
[a piece](https://mlopsfluff.dstack.ai/p/notebooks-and-mlops-choose-one?s=r)

> pointing out how Jupyter Notebooks, while rightfully loved in Data Science work,

Reviewer: ```suggestion
pointing out how Jupyter Notebooks, while rightfully loved in data science work,
```

  shows you how to train and save your models with CML in a two-part series
  using a
  [self-hosted AWS EC2 runner](https://dvc.org/blog/CML-runners-saving-models-1)
  and
  [with CML and DVC on a dedicated AWS EC2 runner](https://dvc.org/blog/CML-runners-saving-models-2)
- The
  [Part 1](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines),
  [Part 2](https://dvc.org/blog/end-to-end-computer-vision-api-part-2-local-experiments)
  and
  [Part 3](https://dvc.org/blog/end-to-end-computer-vision-api-part-3-remote-exp-ci-cd)
  tutorials of [**Alex Kim's**](https://www.linkedin.com/in/alex000kim/)
  End-to-End Computer Vision API project are out and filled with great learning!
- [**Milecia McGregor**](https://twitter.com/FlippedCoding) brings the monthly
  roundup of the Community's best technical questions in our latest
  [Community Gems.](https://dvc.org/blog/april-22-community-gems) 💎

## ✨ Shiny New Docs

We have a [new doc page](https://dvc.org/doc/start/visualization) showcasing the

> new visualizations added to the

Reviewer 2: This page is under an `experiments` subdirectory, i.e. https://dvc.org/doc/start/experiments/visualization

Also, the link would ideally be root-relative so that the advanced local link handling can be used, i.e. `/doc/start/experiments/visualization`

- 📣 PyLadies Berlin is hosting **Doreen**, a data scientist working at
  [Opinary](https://opinary.com/), who will be presenting "Reproducible Machine
  Learning with DVC and Poetry" on May 17th.
  [Join the event here.](https://www.meetup.com/PyLadies-Berlin/events/285313817/)
- 📣 [**Nicolás Eiras**](https://www.linkedin.com/in/nicolas-eiris/) will be
  presenting "Data Versioning: Towards Reproducibility in Machine Learning" at
  [Embedded Vision Summit](https://embeddedvisionsummit.com/2022/session/data-versioning-towards-reproducibility-in-machine-learning/)
  on May 18th in Santa Clara, California.
- 📣 [Montreal PyData](https://www.meetup.com/PyData-MTL/) will host a
  [Meetup](https://www.meetup.com/PyData-MTL/events/285894672/) on June 16th
  with two presentations, "Introduction to Trustworthy Machine Learning for the
  Enterprise" by [**Mohamed Leila**](https://www.linkedin.com/in/mohamedleila/),
  ServiceNow and "ML in production in the video game industry: Ubisoft's use
  case" by
  [**Jean-Michel Daignan**](https://www.linkedin.com/in/jeanmicheldaignan/),
  Ubisoft

## Other Fun Stuff

- [New Awesome list](https://github.com/gaocegege/awesome-open-source-mlops)

> - [New Udemy Course including DVC](https://www.udemy.com/course/dvc-and-git-for-data-science/)

Reviewer 2: This link seems fine, Udemy is just refusing the link checker because it's either blocking bots or requiring logins.

Author: Thanks @rogermparent!!! 🙏
